app/common/ts_util.c

```diff
@@ -0,0 +1,252 @@
+
+#include <petsc.h>
+#include <petscts.h>
+#include <petsc/private/tsimpl.h>
```

Comment:
Is this header part of the petsc installation?

Reply:
> 

Yes